docs/Conceptual/00 - Introduction.md

```diff
@@ -0,0 +1,188 @@
+1. Scalability and density
+    - What are the scalability characteristics of the solution as an individual tenant scales up, and/or the number of tenants scales up?
+1. Portability
+    - Can a tenant be moved (transparently) from one physical resource to another to resolve conflicts and/or as the tenant scales/buys more resources?
+1. Cost
+    - What is the cost model for the resource?
+1. Deployment and Manageability
+    - What are the deployment and manageability implications as the number of tenants scales up? What are the observability and monitoring characteristics for a an individual tenant? For the root tenant?
+1. Versioning and updates
+    - Are there implications for versioning or updating tenanted solutions? What about the cost of updating the management plane for those services?
+
+## Shared infrastructure v. isolated infrastructure
+What about higher-level SaaS services like messaging, document creation, CRM, or telephony? What happens when we aggregate services outside our tenancy host?
+
+The approach to multi-tenancy for those depends somewhat on the architecture of the service providers themselves. Can you reflect your own tenancy hierarchy into those services, or do you treat them as a single shared resource consumed by all your tenants? Even if you can reflect that tenancy hierarchy, should you do so?
+
+This is actually a common decision you need to make for every resource you consume in your multi-tenanted solution. Should we use shared infrastructure or isolated infrastructure?
+
+With shared infrastructure, compute and storage requirements for different tenants are dispatched in common services. Isolation is provided by features within those common services.
+
+For example, you might have a single SQL Server Database instance shared between multiple tenants, whose data is partitioned using Row Level Security. Or a service which abstracts a single instance of Twilio dispatching all the text messages on behalf of all tenants using common credentials supplied by the parent tenant provider.
+
+With isolated infrastructure, each tenant gets its own service instances.
+
+For example each tenant might get its own SQL Server Database Instance, or even its own SQL Server instance. Or a service which dispatches text messages to a particular Twilio account on behalf of a particular tenant, using credentials specific to that tenant.
+
+The ultimate form of this isolated infrastructure is “bring your own service”. In this case, a multi-tenanted provider allows you to configure their solution with your own instances of particular storage or compute infrastructure, encryption keys, credentials etc.
+
+For example, a tenant might configure the text-message dispatch service provided by its parent to use the tenant’s own Twilio account, giving it access to a key in KeyVault that contains their Twilio credentials, configured with access rights appropriate for the service.
```

Comment:
Is there something here that you could have co-mingled data, each encrypted with a tenant specific encryption key?

Reply:
Added an example of this for a "shared infrastructure"